src/datagen/maximumindependentset.jl

```diff
@@ -0,0 +1,33 @@
+using Graphs
+
+struct MaximumIndependentSetGenerator <: AbstractModelGenerator
+    n::Int
+    k::Int
+end
+
+function fill_with_generator!(cpmodel::CPModel, gen::MaximumIndependentSetGenerator;  rng::AbstractRNG = MersenneTwister())
+    graph = Graphs.SimpleGraphs.barabasi_albert(gen.n, gen.k, seed=rand(rng, typemin(Int64):typemax(Int64)))
+
+    # create variables
+    vars = SeaPearl.IntVar[]
+    for v in Graphs.vertices(graph)
+        push!(vars, SeaPearl.IntVar(0, 1, "node_" * string(v), cpmodel.trailer))
+        addVariable!(cpmodel, last(vars))
+    end
+
+    # edge constraints
+    for e in Graphs.edges(graph)
+        SeaPearl.addConstraint!(cpmodel, SeaPearl.SumLessThan([vars[e.src], vars[e.dst]], 1, cpmodel.trailer))
```

Comment:
Yes that's right, it's the index of the nodes in the graph. Since the variables have been defined in the same order as in the node array of the graph, this also corresponds to the node index in `vars`.